crates/sdk/src/migrations.rs

```diff
@@ -431,6 +506,46 @@ impl Display for UpdateStatus {
     }
 }
 
+/// Check if a scheduled migration should take place at this block height.
+/// If so, apply it to the DB.
+pub fn commit<D: DB>(
+    db: &D,
+    height: BlockHeight,
+    mut migration: Option<ScheduledMigration<D::Migrator>>,
+) -> Option<ScheduledMigration<D::Migrator>>
```

Comment:
I'm slightly more in favor of mutating the migrations option in-place (i.e. by passing a `&mut Option<...>` ref), which we could steal with `std::mem::take` or `Option::take`